fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryIndexPlan.java

```diff
@@ -206,14 +208,39 @@ public int getComplexity() {
     @Nonnull
     @Override
     public PlannerGraph rewritePlannerGraph(@Nonnull final List<? extends PlannerGraph> childGraphs) {
-        final PlannerGraph.Node root =
-                new PlannerGraph.Node(this,
-                        getClass().getSimpleName(),
-                        getRange() + "; " + (scanType == IndexScanType.BY_VALUE ? scanType : "") + "; " + (reverse ? "reverse" : ""));
-        final PlannerGraph.SourceNode source = new PlannerGraph.SourceNode(getIndexName());
-        return PlannerGraph.builder(root)
-                .addNode(source)
-                .addEdge(source, root, new PlannerGraph.Edge())
-                .build();
+        final Optional<TupleRange> tupleRangeOptional = getTupleRangeOptional();
+
+        final ImmutableList.Builder<String> detailsBuilder = ImmutableList.builder();
+        final ImmutableMap.Builder<String, Attribute> additionalAttributes = ImmutableMap.builder();
+
+        if (scanType != IndexScanType.BY_VALUE) {
+            detailsBuilder.add("scanType: {{scanType}}");
+            additionalAttributes.put("scanType", Attribute.gml(scanType.toString()));
+        }
+
+        if (tupleRangeOptional.isPresent()) {
+            final TupleRange tupleRange = tupleRangeOptional.get();
+            detailsBuilder.add("range: " + tupleRange.getLowEndpoint().toString(false) + "{{low}}, {{high}}" + tupleRange.getHighEndpoint().toString(true));
+            additionalAttributes.put("low", Attribute.gml(tupleRange.getLow() == null ? "∞" : tupleRange.getLow().toString()));
+            additionalAttributes.put("high", Attribute.gml(tupleRange.getHigh() == null ? "∞" : tupleRange.getHigh().toString()));
+        } else {
+            detailsBuilder.add("comparisons: {{comparisons}}");
+            additionalAttributes.put("comparisons", Attribute.gml(comparisons.toString()));
+        }
+
+        if (reverse) {
+            detailsBuilder.add("direction: {{direction}}");
+            additionalAttributes.put("direction", Attribute.gml("reversed"));
+        }
+
+        return PlannerGraph.fromNodeAndChildGraphs(
+                new PlannerGraph.OperatorNodeWithInfo(this,
+                        NodeInfo.INDEX_SCAN_OPERATOR,
+                        detailsBuilder.build(),
+                        additionalAttributes.build()),
+                ImmutableList.of(
+                        PlannerGraph.fromNodeAndChildGraphs(
+                                new PlannerGraph.DataNodeWithInfo(NodeInfo.INDEX_DATA, ImmutableList.copyOf(getUsedIndexes())),
```

Comment:
As you know, this isn't my favorite representation of index scans. I would prefer to think of an index scan as a single operator with the index name as an attribute, like the predicate for a predicate filter.

I'm happy to merge this for now and continue this discussion another time, though. Can you create an issue to track it?

Reply:
tabled

Reply:
https://github.com/FoundationDB/fdb-record-layer/issues/973